crates/sui-core/src/authority/execution_time_estimator.rs

```diff
@@ -148,19 +221,23 @@ impl ExecutionTimeObserver {
                 .moving_average
                 .add_sample(command_duration);
 
-            // Send a new observation through consensus if our current moving average
-            // differs too much from the last one we shared.
-            // TODO: Consider only sharing observations for entrypoints with congestion.
+            // Send a new observation through consensus if:
+            // - our current moving average differs too much from the last one we shared, and
+            // - the tx has at least one mutable shared object with utilization that's too high
             // TODO: Consider only sharing observations that disagree with consensus estimate.
             let new_average = local_observation.moving_average.get_average();
-            if local_observation
-                .last_shared
-                .is_none_or(|(last_shared, last_shared_timestamp)| {
-                    let diff = last_shared.abs_diff(new_average);
-                    diff > new_average.mul_f64(OBSERVATION_SHARING_DIFF_THRESHOLD)
-                        && last_shared_timestamp.elapsed() > OBSERVATION_SHARING_MIN_INTERVAL
-                })
-            {
+            let diff_exceeds_threshold = {
+                local_observation
+                    .last_shared
+                    .is_none_or(|(last_shared, last_shared_timestamp)| {
+                        let diff = last_shared.abs_diff(new_average);
+                        diff >= new_average.mul_f64(OBSERVATION_SHARING_DIFF_THRESHOLD)
+                            && last_shared_timestamp.elapsed() >= OBSERVATION_SHARING_MIN_INTERVAL
+                    })
+            };
```

Comment:
nit - no need for braces around this expression?

Reply:
fixed thanks